RFS/src/main/java/org/opensearch/migrations/metadata/CreationResult.java

```diff
@@ -0,0 +1,46 @@
+package org.opensearch.migrations.metadata;
+
+import java.util.Optional;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+
+@Builder
+@Data
+public class CreationResult implements Comparable<CreationResult>  {
+    private final String name;
+    private final Exception exception;
+    private final CreationFailureType failureType;
+    public boolean wasSuccessful() {
+        return getFailureType() == null; 
+    }
+
+    public boolean wasFatal() {
+        return Optional.ofNullable(getFailureType()).map(CreationFailureType::isFatal)
+            .orElse(false);
+    }
+
+    @AllArgsConstructor
+    @Getter
+    public static enum CreationFailureType {
```

Comment:
Might be missing a way to show that we did not create an item because it wasn't included on an allow list, curious what others think of these categories

Reply:
i'm good with these categories for now. Maybe to extend to stuff like the allowList, we shouldn't call this a FailureType.. maybe CreationSkipped?